geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaGeoMessage.scala

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2015 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.locationtech.geomesa.kafka
+
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+import java.util
+
+import com.typesafe.scalalogging.slf4j.Logging
+import kafka.message.MessageAndMetadata
+import kafka.producer.KeyedMessage
+import org.joda.time.Instant
+import org.locationtech.geomesa.feature.EncodingOption.EncodingOptions
+import org.locationtech.geomesa.feature.{AvroFeatureDecoder, AvroFeatureEncoder, SimpleFeatureDecoder, SimpleFeatureEncoder}
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+sealed trait KafkaGeoMessage {
+
+  def timestamp: Instant
+}
+
+object KafkaGeoMessage {
+
+  implicit val clock = SystemClock
+
+  def clear()(implicit clock: Clock): Clear =
+    new Clear(clock.now)
+
+  def delete(id: String)(implicit clock: Clock): Delete =
+    new Delete(clock.now, id)
+
+  def createOrUpdate(sf: SimpleFeature)(implicit clock: Clock): CreateOrUpdate =
+    new CreateOrUpdate(clock.now, sf)
+}
+
+/** Create a new simple feature or update an existing [[SimpleFeature]]
+  *
+  * @param feature the [[SimpleFeature]]
+  */
+case class CreateOrUpdate(override val timestamp: Instant, feature: SimpleFeature) extends KafkaGeoMessage {
+
+  val id = feature.getID
+}
+
+/** Delete an existing [[SimpleFeature]]
+  *
+  * @param id the id of the simple feature
+ */
+case class Delete(override val timestamp: Instant, id: String) extends KafkaGeoMessage
+
+/** Delete all [[SimpleFeature]]s
+  */
+case class Clear(override val timestamp: Instant) extends KafkaGeoMessage
+
+trait Clock {
+  def now: Instant
+}
+
+object SystemClock extends Clock {
```

Comment:
Oh, hmm.  Is there a way to work around that in the tests?  I hate to add layers of complexity *just* for testing.